File: myapplication_ui/src/services/AssignedToService.js
import http from "../http-common"; 

class AssignedToService {
  getAllAssignedTos(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/assignedTo/assignedTos`, searchDTO);
  }

  get(assignedToId) {
    return this.getRequest(`/assignedTo/${assignedToId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/assignedTo?field=${matchData}`, null);
  }

  addAssignedTo(data) {
    return http.post("/assignedTo/addAssignedTo", data);
  }

  update(data) {
  	return http.post("/assignedTo/updateAssignedTo", data);
  }
  
  uploadImage(data,assignedToId) {
  	return http.postForm("/assignedTo/uploadImage/"+assignedToId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new AssignedToService();

File: src/main/java/com/myapplication/service/AssignedToService.java
package com.myapplication.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.myapplication.domain.AssignedTo;
import com.myapplication.dto.AssignedToDTO;
import com.myapplication.dto.AssignedToSearchDTO;
import com.myapplication.dto.AssignedToPageDTO;
import com.myapplication.dto.AssignedToConvertCriteriaDTO;
import com.myapplication.service.GenericService;
import com.myapplication.dto.common.RequestDTO;
import com.myapplication.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface AssignedToService extends GenericService<AssignedTo, Integer> {

	List<AssignedTo> findAll();

	ResultDTO addAssignedTo(AssignedToDTO assignedToDTO, RequestDTO requestDTO);

	ResultDTO updateAssignedTo(AssignedToDTO assignedToDTO, RequestDTO requestDTO);

    Page<AssignedTo> getAllAssignedTos(Pageable pageable);

    Page<AssignedTo> getAllAssignedTos(Specification<AssignedTo> spec, Pageable pageable);

	ResponseEntity<AssignedToPageDTO> getAssignedTos(AssignedToSearchDTO assignedToSearchDTO);
	
	List<AssignedToDTO> convertAssignedTosToAssignedToDTOs(List<AssignedTo> assignedTos, AssignedToConvertCriteriaDTO convertCriteria);

	AssignedToDTO getAssignedToDTOById(Integer assignedToId);







}





